### ISSUE TITLE
Allow to reset “Don't Show Again” preference

### ISSUE THREAD

**Issue by OliverJAsh** [microsoft/vscode#24815]
VSCode’s “top banner” often includes a “Don't Show Again” button. However it's not clear how a user should undo this action.

Would it be possible to save these preferences to the user's workspace settings, perhaps? Then it is easy for the user to undo this action.

An example of a user needing this: https://github.com/Microsoft/vscode/issues/23314

**Comment by ajit-id** [comment]
I want it too 🙋

**Comment by bpasero** [comment]
Yes there should be a way to reset any of the messages that have this option.

**Comment by tmknight** [comment]
Any progress on this year old request?

**Comment by joshuaquek** [comment]
I clicked do not show again on an error message and now I dont ever see it anymore. However, I need to see what the error is now. How do I re-enable this error message to show again?

**Comment by DaRockJaeger** [comment]
Yeah! I clickend on "Don't show again", and I never get the message again! I can only solve my problem by
1. deinstall Visual Studio Code,
2. delete C:\Users\[Username]\.vscode
3. delete C:\Users\[Username]\AppData\Roaming\Code 

This is so awkward. Please implement feature where I can just reset my "Don't show it again" settings!

**Comment by yaseralnajjar** [comment]
2 years later... this ain't solved!

I managed to fix this without removing vscode by:

1. Close any opened VSCode and make sure it ain't running in the background
2. Go to C:\Users[Username]\AppData\Roaming\Code\\User
3. Rename globalStorage into globalStorage2

**WARNING: THIS WILL REMOVE ALL GLOBAL SETTINGS YOU SET BEFORE**

**Comment by MrJmpl3** [comment]
The way without lost your settings if using a SQLite Browser one the database in globalStorage  and set dontShowPrompt in false

**Comment by yvess** [comment]
I needed this for the "Reopen in containers" popup in this case the sqlite key is "ms-vscode-remote.remote-containers" and the value is "{"showReopenInContainerNotification":true}" this was set to false. thanks @MrJmpl3

**Comment by Dean-NC** [comment]
Any update ??

**Comment by DatXN** [comment]
I cannot make this work on Mac.
Any solution ?

**Comment by DatXN** [comment]
Thank to MrJmpl3, I be able to reset this on Mac without uninstalling VSCode. Here is how to do it on MAC OS:

1. Quick all VS Code instances (Cmd +Q)
2. Instal SQL Lite browser https://sqlitebrowser.org
3. Go to: /Users/[user_name]/Library/Application Support/Code/User/workspaceStorage/ => Find the one that associate with your workspace that you previously clicked "Don't show this promt again" (by looking into workspace.json and identify the name)
4. Open stage.vscdb file with SQL Lite
5. Look for data key "ms-vscode.csharp" 
6. Edit _"assetPromptDisabled":true_ to  _"assetPromptDisabled":**false**_
7. Save change
8. Reopen vs code

<img width="196" alt="image" src="https://user-images.githubusercontent.com/5478040/67828790-56232f00-fb07-11e9-87d1-ee89c0eac80d.png">

**Comment by AlJohri** [comment]
thanks @yvess, that worked for me. I opened `~/Library/Application\ Support/Code/User/globalStorage/state.vscdb` using TablePlus and was able to change `showReopenInContainerNotification` from false to true.

there should definitely be an easier way to do this!

**Comment by amaljose02** [comment]
Hi, I was able to do this on Windows without uninstalling VS Code. it's pretty similar to the one said by @DatXN on MAC OS.

Here is how I did it:
1. Quit all VS Code instances
2. Instal SQL Lite browser [https://sqlitebrowser.org](https://sqlitebrowser.org)
3. Go to C:\Users\[user_name]\AppData\Roaming\Code\User\globalStorage
4. Open state.vscdb file with SQLite
5. Look for data key integrityService
6. Edit "_dontShowPrompt":true_ to "_dontShowPrompt":false_
7. Save change
8. Reopen vs code

![vscode](https://user-images.githubusercontent.com/31209960/73709831-31eb7580-4728-11ea-942b-f5e1f2dbea38.png)

**Comment by onefork** [comment]
For Windows the path to the SQL Lite file is `%AppData%\Code\User\globalStorage\state.vscdb`.

> Hi, I was able to do this on Windows without uninstalling VS Code. it's pretty similar to the one said by @DatXN on MAC OS.
> 
> Here is how I did it:
> 
> 1. Quit all VS Code instances
> 2. Instal SQL Lite browser https://sqlitebrowser.org
> 3. Go to C:\Users[user_name]\AppData\Roaming\Code\User\globalStorage
> 4. Open state.vscdb file with SQLite
>

**Comment by nihaals** [comment]
Is there an easy way of finding which directory in `User/workspaceStorage` a workspace corresponds to?

**Comment by raulshma** [comment]
@nihaals I don't know of a proper way but this will work. Open the workspace change a file and then sort by modified in the workspaceStorage folder.

**Comment by godot11** [comment]
[bump] as I believe there should be at least an in-app way to undo this. At least some commands like `ShowDisabledNotifications`, or even just a `ResetDisabledNotifications` that forgets all "don't show again' settings. 

I accidentally misclicked on a _"There are extensions that can help opening ".png" files..." type of notification, and now I won't ever find out what were they unless I spend time installing an SQL editor and fiddle around with obscure files at obscure locations, possibly breaking my VS Code setup in the progress. 

It's a very annoying very tiny thing.

**Comment by nihaals** [comment]
There is also a trend of more things moving to the workspace database from settings, for example `python.pythonPath`, so having a way to edit these values is even more important

**Comment by GitMensch** [comment]
Can the mentioned workarounds be part of an official FAQ? Something like "things we don't plan to implement but you may want to do it by adjusting the used sqlite database following the steps mentioned here?
